cpp/graph_util_module/algorithms/topological_sort.hpp

```diff
@@ -0,0 +1,60 @@
+#include <list>
+
+#include <mgp.hpp>
+
+void TopologicalSort(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  try {
+    mgp::memory = memory;
+    const auto record_factory = mgp::RecordFactory(result);
+    const auto graph = mgp::Graph(memgraph_graph);
+
+    std::map<mgp::Node, uint64_t> in_degrees;
+
+    for (const auto node : graph.Nodes()) {
+      in_degrees[node] = 0;
+    }
+
+    for (const auto node : graph.Nodes()) {
+      for (const auto relationship : node.OutRelationships()) {
+        const auto to_node = relationship.To();
```

Comment:
You can use for this part Graph::Relationships(). Maybe check this part our as it iterates over `outRelationships` for all nodes in graph

Reply:
WIll do.